### PR TITLE
Add support for EventGrid bindings

### PR DESCRIPTION
**Pull request by elprans**

### CODE REVIEW

File: azure/worker/bindings/eventgrid.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import datetime
+import typing
+
+from azure.functions import _abc as azf_abc
+
+from . import meta
+from .. import protos
+
+
+class GridEvent(azf_abc.GridEvent):
```
Comment (pragnagopa) [inline azure/worker/bindings/eventgrid.py]: EventGrid instead of GridEvent.

File: tests/eventgrid_functions/eventgrid_trigger/function.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "scriptFile": "__init__.py",
+  "disabled": false,
+
+  "bindings": [
+    {
+      "authLevel": "anonymous",
```
Comment (pragnagopa) [inline tests/eventgrid_functions/eventgrid_trigger/function.json]: remove authlevel

File: tests/test_eventgrid_functions.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                "storageDiagnostics": {
+                    "batchId": "b4229b3a-4d50-4ff4-a9f2-039ccf26efe9"
+                }
+            },
+            "dataVersion": "",
+            "metadataVersion": "1"
+        }]
+
+        r = self.request('POST', 'eventgrid_trigger', json=data)
+        self.assertEqual(r.status_code, 200)
```
Comment (pragnagopa) [inline tests/test_eventgrid_functions.py]: From the doucmentation on how to [manually-post-the-request](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-grid#manually-post-the-request), you need Set an aeg-event-type: Notification header.

File: tests/eventgrid_functions/ping/function.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "scriptFile": "main.py",
+  "disabled": false,
+  "bindings": [
+    {
+      "authLevel": "anonymous",
```
Comment (pragnagopa) [inline tests/eventgrid_functions/ping/function.json]: Can you open an issue to track figuring out testing with different autheLevels? anonymous is not the default.  Eventually, we need to remove "authLevel": "anonymous" from all the http tests

File: azure/worker/testutils.py [hunk 5]
```diff
@@ -550,13 +552,26 @@ def _main():
 
     args = parser.parse_args()
 
+    extensions = pathlib.Path(args.scriptroot) / 'bin'
+
+    if not extensions.exists():
+        if extensions.is_symlink():
+            extensions.unlink()
+
+        extensions.symlink_to(EXTENSIONS_PATH, target_is_directory=True)
```
Comment (pragnagopa) [inline azure/worker/testutils.py]: add logs to indicate which extensions are installed.

File: azure/functions/_abc.py [hunk 6]
```diff
@@ -199,6 +199,42 @@ def pop_receipt(self) -> typing.Optional[str]:
         pass
 
 
+class GridEvent(abc.ABC):
```
Comment (pragnagopa) [inline azure/functions/_abc.py]: I think better name would be EventGridEvent

**Comment by asavaritayal** [comment]
@elprans - Any update on this? Can you please take a look?

**Pull request merged**